cmake/firebase_unity_version.cmake

```diff
@@ -27,7 +27,7 @@ set(FIREBASE_UNITY_JAR_RESOLVER_VERSION "1.2.169"
 )
 
 # https://github.com/firebase/firebase-cpp-sdk
-set(FIREBASE_CPP_SDK_PRESET_VERSION "v8.8.0"
+set(FIREBASE_CPP_SDK_PRESET_VERSION "v8.9.0"
```

Comment:
With this change, I think we also need to change the includes for the mutex header, since that got changed with 8.9.0

Reply:
It's done in this PR https://github.com/firebase/firebase-unity-sdk/pull/142